### ISSUE TITLE
"bower: command not found"

### ISSUE THREAD

**Issue by jjg1914** [nikhilmodak/gulp-ngdocs#41]
Linking the issue I opened with the NPM team for reference: [link](https://github.com/npm/npm/issues/7722)

When gulp-ngdocs gets installed side-by-side bower, bower is not installed in the node_modules of the gulp-ngdocs and the install script breaks.

Could the install script be removed and the needed files vendored in the repo instead?

**Comment by lukewestby** [comment]
+1

If this isn't feasible, could bower be called from the node_modules/.bin directory directly?

**Comment by andyperlitch** [comment]
I'd really like to avoid checking in vendor files. @lukewestby I don't know if this will work, but I'll try it out when I have the chance.

**Comment by andyperlitch** [comment]
@lukewestby I ran a preliminary test and it appears to work this way. Basically we would need to list bower as a dependency , then change the build to call `./node_modules/.bin/bower install` instead of `bower install`.

However, we should think about some other consequences, like if it reads the .bowerrc file.

**Comment by iyel** [comment]
+1

**Comment by MikaAK** [comment]
+1 I'm having issues when running as a root user (inside of docker it's hard to get a non-sudo user setup with node) and having the `bower install` causes a `ESUDO Error` as it's run with root. Since you can't port `--allow-root` over from the `npm` command can it be added to the bower command or replace using bower internally? It seems a bit odd to have a hard requirement on bower.

**Comment by DigTheDoug** [comment]
+1
It's unfortunate, this is breaking our build server. What about using the npm versions of those bower packages instead and not relying on bower?

**Comment by stringbeans** [comment]
+1 same issue as @MikaAK

**Comment by jjg1914** [comment]
This looks fixed as of v0.2.13, at least for me. Can anyone else confirm?

**Comment by pioug** [comment]
Yes it is fixed in v0.2.13. Installation runs well without global `bower`. I had issue previously with v0.2.10.
